src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -93,6 +94,8 @@ public CSharpType(TypeProvider implementation, string ns, string name, bool isVa
         public TypeProvider Implementation => _implementation ?? throw new InvalidOperationException($"Not implemented type: '{Namespace}.{Name}'");
         public bool IsNullable { get; }
 
+        public WhereExpression IsType(CSharpType type) => new WhereExpression(this, type);
```

Comment:
I do not think this method belong to this class.
We should add it to `Snippets.Where` since it is building a where expression, for example:
```csharp
public WhereExpression Inherits(CSharpType type, CSharpType inherits)
```
or we could make a `params` and call it `Implements`